src/2024h2/parallel-front-end.md

```diff
@@ -0,0 +1,65 @@
+# Stabilize Parallel Front End
+
+| Metadata |                           |
+| -------- |---------------------------|
+| Owner(s) | [@SparrowLii]             |
+| Teams    | [Parallel Rustc WG] |
+| Status   | Existing problems need to be resolved to stabilize                          |
+
+## Motivation
+
+The parallel front end has been implemented in nightly, but there are still many problems that prevent it from being stable and used at scale.
+
+### The status quo
+
+Many current [issues] reflect ICE or deadlock problems that occur during the use of parallel front-ends. We need to resolve these issues to ensure functional correctness.
+
+The existing compiler testing framework is not sufficient for the parallel front end, and we need to enhance it to ensure the correct functionality of the parallel front end.
+
+The current parallel front end still has room for further improvement in compilation performance, such as parallelization of HIR lowering and macro expansion, and reduction of data contention under more threads (>= 16).
+
+We can use parallel front end in bootstrap to alleviate the problem of slow build of the whole project.
+
+### The next 6 months
+
+- Solve most of the existing issues (unless they are due to lack of rare hardware environment or are almost impossible to reproduce).
```

Comment:
These are good goals, but I wonder if they are perhaps a bit too ambitious for the next 6 months. Maybe we should first focus on better testing, bootstrap integration, and stabilization. And then later on more performance improvements?

Reply:
It also depends on many resources we expect to put into this goal, if we know who and how much people are expecting to be participating

Reply:
Yea, In fact, the biggest problem is that many issues are difficult to reproduce. I think we should at least solve most of the ICE and deadlock issues